### PR TITLE
Add basic GH workflow

### PR DESCRIPTION
**Pull request by jamilbk**
Just laying down some boilerplate in an attempt to get a rudimentary CI pipeline started.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+jobs:
+  cargo_windows:
+    name: Check
+    strategy:
+      matrix:
+        rust: [stable, nightly]
+        runs-on:
+          - windows-2019
+          - windows-2022
```
Comment (conectado) [inline .github/workflows/build.yml:27]: I think we would like more OSes here

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        uses: actions/checkout@v3
+      - name: Update toolchain
+        run: rustup update --no-self-update ${{ matrix.rust }} && rustup default ${{ matrix.rust }}
+      - name: Authenticate SSH agent
+        run: |
+          eval `ssh-agent -s`
+          ssh-add - <<< '${{ secrets.NDISAPI_CLONE_KEY }}'
+          ssh-add - <<< '${{ secrets.VIRTNET_CLONE_KEY }}'
+      - name: Run cargo check
+        run: cargo check
```
Comment (conectado) [inline .github/workflows/build.yml]: We want also:
`cargo clippy -- -D clippy::all` and `cargo test`

File: .github/workflows/build.yml (hunk 1 above)
Comment (jamilbk) [reply .github/workflows/build.yml:27]: Added

File: .github/workflows/build.yml (hunk 2 above)
Comment (jamilbk) [reply .github/workflows/build.yml]: Added

**Review by conectado — APPROVED** [review]

**Pull request merged**